SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SchemaGenerator.h

```diff
@@ -11,12 +11,13 @@ class FCodeWriter;
 struct FComponentIdGenerator;
 
 extern TArray<UClass*> SchemaGeneratedClasses;
-extern TMap<FString, FSchemaData> ClassPathToSchemaData;
+extern TMap<FString, FActorSchemaData> ActorClassPathToSchema;
+extern TMap<FString, FSubobjectSchemaData> SubobjectClassPathToSchema;
 extern TMap<FString, uint32> LevelPathToComponentId;
 
 // Generates a schema file, given an output code writer, component ID, Unreal type and type info.
 void GenerateActorSchema(FComponentIdGenerator& IdGenerator, UClass* Class, TSharedPtr<FUnrealType> TypeInfo, FString SchemaPath);
-void GenerateSubobjectSchema(UClass* Class, TSharedPtr<FUnrealType> TypeInfo, FString SchemaPath);
-void GenerateSubobjectSchemaForActor(FComponentIdGenerator& IdGenerator, UClass* ActorClass, TSharedPtr<FUnrealType> TypeInfo, FString SchemaPath, FSchemaData& ActorSchemaData);
-FSubobjectSchemaData GenerateSubobjectSpecificSchema(FCodeWriter& Writer, FComponentIdGenerator& IdGenerator, FString PropertyName, TSharedPtr<FUnrealType>& TypeInfo, UClass* ComponentClass);
-void GenerateActorIncludes(FCodeWriter& Writer, TSharedPtr<FUnrealType>& TypeInfo);
+void GenerateSubobjectSchema(FComponentIdGenerator& IdGenerator, UClass* Class, TSharedPtr<FUnrealType> TypeInfo, FString SchemaPath);
```

Comment:
Could these have more descriptive names, or comments? I realise that some of these were here before but it's hard to tell what the different between GenerateSubojectSchema and GenerateSubobjectSpecificSchema is.

Reply:
For the shared pointers, are we passing to something that co-owns the data, or do they just need a reference while they run. It's a little confusing to look at because I'm not sure where the ownership is meant to lie. Reference counting also has to happen on every copy, but that is probably not a big issue.